[🐞 Bug] 同一个session，创建临时表后再查询会提示不存在。

### 数据库类型和版本

MySQL 5.7、8.0

### 问题描述

如下SQL，整体执行没有问题，分开执行，临时表会提示不存在。
DROP TEMPORARY TABLE IF EXISTS tmp_t1;

CREATE TEMPORARY TABLE tmp_t1(
id INT NOT NULL AUTO_INCREMENT PRIMARY KEY
,name VARCHAR(100)
);

 INSERT tmp_t1(name) VALUES('test');

 SELECT *
 FROM tmp_t1
 LIMIT 100;

### 错误日志

```shell
DBX debug log
Exported: 2026-06-21T01:17:19.849Z
User agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko)
Platform: MacIntel
Timezone: Asia/Shanghai

[2026-06-21T01:16:59.548Z] [INFO] [DBX][debug-log] enabled {"url":"tauri://localhost","viewport":"1470x860","devicePixelRatio":2,"language":"zh-CN","online":true}
[2026-06-21T01:16:59.548Z] [DEBUG] [DBX][api:start] saveDesktopSettings
[2026-06-21T01:16:59.564Z] [DEBUG] [DBX][api:success] {"operation":"saveDesktopSettings","elapsedMs":16}
[2026-06-21T01:17:03.815Z] [DEBUG] [ui:click] {"tag":"button"}
[2026-06-21T01:17:06.998Z] [DEBUG] [ui:click] {"tag":"button","label":"执行"}
[2026-06-21T01:17:06.999Z] [INFO] [DBX][executeTabSql:start] {"traceId":"9f220e61","tabId":"848f797f-11e6-4559-8477-e8dd064b807b","mode":"query","connectionId":"4ec0c1db-f86c-40d3-86f1-e79f9b1a844e","database":"testdb","sql":"DROP TEMPORARY TABLE IF EXISTS tmp_t1;\n\nCREATE TEMPORARY TABLE tmp_t1(\nid INT NOT NULL AUTO_INCREMENT PRIMARY KEY\n,name VARCHAR(100)\n);\n\n INSERT tmp_t1(name) VALUES('test');\n\n SELECT *\n FROM tmp_t1\n LIMIT 100;"}
[2026-06-21T01:17:07.043Z] [DEBUG] [DBX][api:start] prepareQueryPaginationExecutionPlan
[2026-06-21T01:17:07.053Z] [DEBUG] [DBX][api:success] {"operation":"prepareQueryPaginationExecutionPlan","elapsedMs":10}
[2026-06-21T01:17:07.053Z] [INFO] [DBX][executeTabSql:execute-multi:start] {"traceId":"9f220e61","elapsed":"54ms"}
[2026-06-21T01:17:07.053Z] [DEBUG] [DBX][api:start] executeMulti
[2026-06-21T01:17:07.061Z] [DEBUG] [DBX][api:success] {"operation":"executeMulti","elapsedMs":8}
[2026-06-21T01:17:07.061Z] [INFO] [DBX][executeTabSql:execute-multi:done] {"traceId":"9f220e61","resultCount":4,"rowCounts":[0,0,0,1],"columnCounts":[0,0,0,2],"elapsed":"62ms"}
[2026-06-21T01:17:07.061Z] [INFO] [DBX][executeTabSql:result:assigned] {"traceId":"9f220e61","activeResultIndex":3,"rowCount":1,"columnCount":2,"backendMs":0,"elapsed":"62ms"}
[2026-06-21T01:17:07.061Z] [INFO] [DBX][executeTabSql:metadata:start] {"traceId":"9f220e61","elapsed":"62ms"}
[2026-06-21T01:17:07.061Z] [INFO] [DBX][executeTabSql:metadata:editability:start] {"traceId":"9f220e61","elapsed":"62ms"}
[2026-06-21T01:17:07.061Z] [DEBUG] [DBX][api:start] analyzeEditableQueryEditability
[2026-06-21T01:17:07.061Z] [INFO] [DBX][executeTabSql:finish] {"traceId":"9f220e61","elapsed":"62ms"}
[2026-06-21T01:17:07.061Z] [INFO] [DBX][ContentArea:result:observed] {"tabId":"848f797f-11e6-4559-8477-e8dd064b807b","rowCount":1,"columnCount":2,"backendMs":0,"isExecuting":false}
[2026-06-21T01:17:07.062Z] [INFO] [DBX][DataGrid:setup] {"traceId":"107f010d","cacheKey":"848f797f-11e6-4559-8477-e8dd064b807b-current-3","rowCount":1,"columnCount":2,"backendMs":0,"loading":false}
[2026-06-21T01:17:07.062Z] [INFO] [DBX][DataGrid:result:prop] {"traceId":"107f010d","cacheKey":"848f797f-11e6-4559-8477-e8dd064b807b-current-3","rowCount":1,"columnCount":2,"backendMs":0,"loading":false,"elapsedSinceSetup":"0ms"}
[2026-06-21T01:17:07.063Z] [INFO] [DBX][DataGrid:display-items:ready] {"traceId":"107f010d","cacheKey":"848f797f-11e6-4559-8477-e8dd064b807b-current-3","displayItemCount":1,"sourceRowCount":1,"elapsedSinceSetup":"1ms"}
[2026-06-21T01:17:07.080Z] [INFO] [DBX][ContentArea:result:nextTick] {"tabId":"848f797f-11e6-4559-8477-e8dd064b807b","elapsed":"19ms","isExecuting":false}
[2026-06-21T01:17:07.080Z] [INFO] [DBX][DataGrid:result:nextTick] {"traceId":"107f010d","cacheKey":"848f797f-11e6-4559-8477-e8dd064b807b-current-3","elapsed":"18ms","loading":false}
[2026-06-21T01:17:07.087Z] [DEBUG] [DBX][api:start] saveHistory
[2026-06-21T01:17:07.087Z] [DEBUG] [DBX][api:start] deleteSchemaCachePrefix
[2026-06-21T01:17:07.087Z] [DEBUG] [DBX][api:success] {"operation":"analyzeEditableQueryEditability","elapsedMs":26}
[2026-06-21T01:17:07.087Z] [INFO] [DBX][executeTabSql:metadata:editability:done] {"traceId":"9f220e61","editable":false,"reason":"not-select","elapsed":"88ms"}
[2026-06-21T01:17:07.087Z] [INFO] [DBX][executeTabSql:metadata:done] {"traceId":"9f220e61","elapsed":"88ms"}
[2026-06-21T01:17:07.088Z] [INFO] [DBX][ContentArea:result:first-frame] {"tabId":"848f797f-11e6-4559-8477-e8dd064b807b","elapsed":"27ms","isExecuting":false}
[2026-06-21T01:17:07.088Z] [INFO] [DBX][DataGrid:result:first-frame] {"traceId":"107f010d","cacheKey":"848f797f-11e6-4559-8477-e8dd064b807b-current-3","elapsed":"26ms","loading":false}
[2026-06-21T01:17:07.088Z] [INFO] [DBX][DataGrid:display-items:first-frame] {"traceId":"107f010d","cacheKey":"848f797f-11e6-4559-8477-e8dd064b807b-current-3","displayItemCount":1,"renderedRows":0,"elapsed":"25ms","elapsedSinceSetup":"26ms","loading":false}
[2026-06-21T01:17:07.097Z] [DEBUG] [DBX][api:success] {"operation":"saveHistory","elapsedMs":10}
[2026-06-21T01:17:07.098Z] [DEBUG] [DBX][api:success] {"operation":"deleteSchemaCachePrefix","elapsedMs":10}
[2026-06-21T01:17:07.098Z] [DEBUG] [DBX][api:start] deleteSchemaCachePrefix
[2026-06-21T01:17:07.098Z] [DEBUG] [DBX][api:start] deleteSchemaCachePrefix
[2026-06-21T01:17:07.133Z] [DEBUG] [DBX][api:start] saveSchemaCache
[2026-06-21T01:17:07.162Z] [DEBUG] [DBX][api:success] {"operation":"deleteSchemaCachePrefix","elapsedMs":64}
[2026-06-21T01:17:07.162Z] [DEBUG] [DBX][api:success] {"operation":"deleteSchemaCachePrefix","elapsedMs":64}
[2026-06-21T01:17:07.162Z] [DEBUG] [DBX][api:success] {"operation":"saveSchemaCache","elapsedMs":29}
[2026-06-21T01:17:10.316Z] [DEBUG] [ui:click] {"tag":"button"}
[2026-06-21T01:17:10.316Z] [INFO] [DBX][executeTabSql:start] {"traceId":"f5b22b3c","tabId":"848f797f-11e6-4559-8477-e8dd064b807b","mode":"query","connectionId":"4ec0c1db-f86c-40d3-86f1-e79f9b1a844e","database":"testdb","sql":"SELECT *\n FROM tmp_t1\n LIMIT 100;"}
[2026-06-21T01:17:10.320Z] [DEBUG] [DBX][api:start] prepareQueryPaginationExecutionPlan
[2026-06-21T01:17:10.323Z] [DEBUG] [DBX][api:success] {"operation":"prepareQueryPaginationExecutionPlan","elapsedMs":3}
[2026-06-21T01:17:10.323Z] [INFO] [DBX][executeTabSql:execute-multi:start] {"traceId":"f5b22b3c","elapsed":"7ms"}
[2026-06-21T01:17:10.323Z] [DEBUG] [DBX][api:start] executeMulti
[2026-06-21T01:17:10.326Z] [ERROR] [DBX][api:error] {"operation":"executeMulti","elapsedMs":3,"error":"Server error: `ERROR 42S02 (1146): Table 'testdb.tmp_t1' doesn't exist'"}
[2026-06-21T01:17:10.326Z] [ERROR] [DBX][executeTabSql:error] {"traceId":"f5b22b3c","elapsed":"10ms","error":"Server error: `ERROR 42S02 (1146): Table 'testdb.tmp_t1' doesn't exist'"}
[2026-06-21T01:17:10.326Z] [INFO] [DBX][executeTabSql:finish] {"traceId":"f5b22b3c","elapsed":"10ms"}
[2026-06-21T01:17:10.327Z] [INFO] [DBX][ContentArea:result:observed] {"tabId":"848f797f-11e6-4559-8477-e8dd064b807b","rowCount":1,"columnCount":1,"backendMs":0,"isExecuting":false}
[2026-06-21T01:17:10.328Z] [INFO] [DBX][DataGrid:setup] {"traceId":"1178e3a9","cacheKey":"848f797f-11e6-4559-8477-e8dd064b807b-current-0","rowCount":1,"columnCount":1,"backendMs":0,"loading":false}
[2026-06-21T01:17:10.328Z] [INFO] [DBX][DataGrid:result:prop] {"traceId":"1178e3a9","cacheKey":"848f797f-11e6-4559-8477-e8dd064b807b-current-0","rowCount":1,"columnCount":1,"backendMs":0,"loading":false,"elapsedSinceSetup":"0ms"}
[2026-06-21T01:17:10.329Z] [INFO] [DBX][DataGrid:display-items:ready] {"traceId":"1178e3a9","cacheKey":"848f797f-11e6-4559-8477-e8dd064b807b-current-0","displayItemCount":1,"sourceRowCount":1,"elapsedSinceSetup":"1ms"}
[2026-06-21T01:17:10.334Z] [INFO] [DBX][ContentArea:result:nextTick] {"tabId":"848f797f-11e6-4559-8477-e8dd064b807b","elapsed":"7ms","isExecuting":false}
[2026-06-21T01:17:10.334Z] [INFO] [DBX][DataGrid:result:nextTick] {"traceId":"1178e3a9","cacheKey":"848f797f-11e6-4559-8477-e8dd064b807b-current-0","elapsed":"6ms","loading":false}
[2026-06-21T01:17:10.334Z] [DEBUG] [DBX][api:start] saveHistory
[2026-06-21T01:17:10.338Z] [INFO] [DBX][ContentArea:result:first-frame] {"tabId":"848f797f-11e6-4559-8477-e8dd064b807b","elapsed":"11ms","isExecuting":false}
[2026-06-21T01:17:10.338Z] [INFO] [DBX][DataGrid:result:first-frame] {"traceId":"1178e3a9","cacheKey":"848f797f-11e6-4559-8477-e8dd064b807b-current-0","elapsed":"10ms","loading":false}
[2026-06-21T01:17:10.338Z] [INFO] [DBX][DataGrid:display-items:first-frame] {"traceId":"1178e3a9","cacheKey":"848f797f-11e6-4559-8477-e8dd064b807b-current-0","displayItemCount":1,"renderedRows":0,"elapsed":"9ms","elapsedSinceSetup":"10ms","loading":false}
[2026-06-21T01:17:10.343Z] [DEBUG] [DBX][api:success] {"operation":"saveHistory","elapsedMs":9}
[2026-06-21T01:17:12.284Z] [DEBUG] [ui:click] {"tag":"button"}
[2026-06-21T01:17:14.095Z] [DEBUG] [ui:click] {"tag":"button","type":"button","label":"外观"}
[2026-06-21T01:17:19.848Z] [DEBUG] [ui:click] {"tag":"button","type":"button","label":"下载日志"}

===== Native / Tauri logs =====
Native log dir: /Users/ken/Library/Logs/com.dbx.app

===== DBX.log =====
[2026-06-21][01:17:07][dbx_lib::commands::query][INFO] [query][execute_multi:start] trace_id=9f220e61-c089-43aa-bc0e-0f17b70ec8ef connection_id=4ec0c1db-f86c-40d3-86f1-e79f9b1a844e database=testdb schema=None sql=DROP TEMPORARY TABLE IF EXISTS tmp_t1;

CREATE TEMPORARY TABLE tmp_t1(
id INT NOT NULL AUTO_INCREMENT PRIMARY KEY
,name VARCHAR(100)
);

 INSERT tmp_t1(name) VALUES('test');

 SELECT *
 FROM tmp_t1
 LIMIT 100;
[2026-06-21][01:17:07][dbx_lib::commands::query][INFO] [query][execute_multi:done] trace_id=9f220e61-c089-43aa-bc0e-0f17b70ec8ef elapsed_ms=6 result_count=4 row_counts=[0, 0, 0, 1] backend_execution_times_ms=[0, 3, 0, 0]
[2026-06-21][01:17:10][dbx_lib::commands::query][INFO] [query][execute_multi:start] trace_id=f5b22b3c-2642-4a67-bb58-894607ce7633 connection_id=4ec0c1db-f86c-40d3-86f1-e79f9b1a844e database=testdb schema=None sql=SELECT *
 FROM tmp_t1
 LIMIT 100;
[2026-06-21][01:17:10][dbx_lib::commands::query][ERROR] [query][execute_multi:error] trace_id=f5b22b3c-2642-4a67-bb58-894607ce7633 elapsed_ms=2 error=Server error: `ERROR 42S02 (1146): Table 'testdb.tmp_t1' doesn't exist'
```

### DBX 版本

0.5.36

### 操作系统

macOS

### 优先级（是否紧急）

🔴 P0 紧急 · 下个版本必须修

### 补充信息

_No response_

--- a/crates/dbx-core/src/db/mysql.rs
+++ b/crates/dbx-core/src/db/mysql.rs
@@ -417,9 +417,13 @@ fn create_pool(url: &str, ca_cert_path: Option<&str>, max_connections: usize) ->
         mysql_async::Opts::from_url(&mysql_async_url(&tls_url.url)).map_err(|e| format!("Invalid MySQL URL: {e}"))?;
     let base_ssl_opts = opts.ssl_opts().cloned();
     let max_connections = max_connections.max(1);
+    // Single-connection pools (max_connections == 1) are client session pools that
+    // must preserve session state (e.g. TEMPORARY TABLEs) across queries.
+    // Disable COM_RESET_CONNECTION for these pools to avoid clearing that state.
     let pool_opts = mysql_async::PoolOpts::new()
         .with_constraints(mysql_async::PoolConstraints::new(1, max_connections).unwrap())
-        .with_inactive_connection_ttl(Duration::from_secs(300));
+        .with_inactive_connection_ttl(Duration::from_secs(300))
+        .with_reset_connection(max_connections > 1);
     let mut builder = mysql_async::OptsBuilder::from_opts(opts)
         .stmt_cache_size(0)
         .prefer_socket(false)

